### PR TITLE
take group creation result into consideration

### PR DESCRIPTION
**Pull request by blizzz**
by catch. At the moment, we assume that createGroup simply succeeds :man_shrugging: Not anymore, comes with unit test.

I wouldn't necessarily port it back (not without a reported bug) as it affects behaviour.

### CODE REVIEW

File: lib/private/Group/Manager.php [hunk 1]
```diff
@@ -210,21 +210,22 @@ public function groupExists($gid) {
...
-			$this->emit('\OC\Group', 'preCreate', array($gid));
+			$this->emit('\OC\Group', 'preCreate', [$gid]);
 			foreach ($this->backends as $backend) {
-				if ($backend->implementsActions(\OC\Group\Backend::CREATE_GROUP)) {
-					$backend->createGroup($gid);
-					$group = $this->getGroupObject($gid);
-					$this->emit('\OC\Group', 'postCreate', array($group));
-					return $group;
+				if ($backend->implementsActions(Backend::CREATE_GROUP)) {
+					if($backend->createGroup($gid)) {
```
Comment (skjnldsv) [inline lib/private/Group/Manager.php]:
```suggestion
					if ($backend->createGroup($gid)) {
```
:see_no_evil:

**Review by skjnldsv — APPROVED** [review]

File: lib/private/Group/Manager.php [hunk 2]
```diff
@@ -210,21 +210,22 @@ public function groupExists($gid) {
 
 	/**
 	 * @param string $gid
-	 * @return \OC\Group\Group
+	 * @return IGroup|bool|null
 	 */
 	public function createGroup($gid) {
 		if ($gid === '' || $gid === null) {
 			return false;
```
Comment (nickvergessen) [inline lib/private/Group/Manager.php]: Should we unify this to null?

Comment (blizzz) [reply lib/private/Group/Manager.php]: the interface's PHPdoc actually only wants an IGroup returned. 

* the group create occ command relies on IGroup
* and the user add command in some way, too
* setup relies on it, too

File: lib/private/Setup.php [hunk 3]
```diff
@@ -380,7 +381,9 @@ public function install($options) {
 			$config->setAppValue('core', 'vendor', $this->getVendor());
 
 			$group =\OC::$server->getGroupManager()->createGroup('admin');
-			$group->addUser($user);
+			if($group instanceof IGroup) {
```
Comment (nickvergessen) [inline lib/private/Setup.php]:
```suggestion
			if ($group instanceof IGroup) {
```

**Comment by blizzz** [comment]
bump

File: core/Command/Group/Add.php [hunk 4]
```diff
@@ -68,6 +68,10 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 			return 1;
 		} else {
 			$group = $this->groupManager->createGroup($gid);
+			if($group === false) {
```
Comment (icewind1991) [inline core/Command/Group/Add.php]: `null` is used as error value here, not `false`

(`!$group instanceof ...` would also be a clearer option here imo)

**Review by icewind1991 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**